Description: fix shrinking menu problem

 The original author of this patch Jiahua Huang[1]. We just modified
 it to later version of the indicator-applet.

 [1]: https://bugs.launchpad.net/indicator-applet/+bug/965953

Author: Tuomas Räsänen <tuomasjjrasanen@opinsys.fi>

---

Origin: Jiahua Huang, <https://launchpadlibrarian.net/105518628/indicator-applet-scroll-arrows.patch>
Bug-Ubuntu: https://launchpad.net/bugs/965953

--- indicator-applet-12.10.1.orig/src/applet-main.c
+++ indicator-applet-12.10.1/src/applet-main.c
@@ -365,6 +365,21 @@ accessible_desc_update (IndicatorObject
   return;
 }
 
+static gboolean
+do_menu_reposition(gpointer menu)
+{
+  gtk_widget_set_size_request(menu, -1, -1);
+  g_timeout_add(20, (GSourceFunc) gtk_menu_reposition, (gpointer) menu);
+  return 0;
+}
+
+static void
+on_menuitem_show (GtkMenuItem *menuitem, gpointer menu)
+{
+  gtk_widget_set_size_request(menu, -1, 350);
+  g_idle_add((GSourceFunc) do_menu_reposition, (gpointer) menu);
+}
+
 static GtkWidget*
 create_menuitem (IndicatorObject * io, IndicatorObjectEntry * entry, GtkWidget * menubar)
 {
@@ -412,6 +427,7 @@ create_menuitem (IndicatorObject * io, I
 
   if (entry->menu != NULL) {
     gtk_menu_item_set_submenu(GTK_MENU_ITEM(menuitem), GTK_WIDGET(entry->menu));
+    g_signal_connect(G_OBJECT(menuitem), "activate", G_CALLBACK(on_menuitem_show), entry->menu);
   }
 
   place_in_menu(menubar, menuitem, io, entry);
